force-app/main/default/classes/SOQL.cls

```diff
@@ -345,17 +443,30 @@ public with sharing class SOQL {
      */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
+        Opportunity[] oppsToUpdate = getOpportunityDetailsForClosedWon();
         // Create a set of Account Ids from the Opportunities
-
+        Set<Id> accountIds = new Set<Id>();
+        for(Opportunity currOpp : oppsToUpdate) {
+            accountIds.add(currOpp.AccountId);
+        }
         // Query the Contacts
-
+        Map<Id, Contact> accToContactMap = new map<Id, Contact>();
         // Loop through each Contact and add the Account Id and Contact to the map
+        Account[] getContacts = [SELECT Id, (SELECT Id, Name, AccountId, FirstName FROM Contacts ORDER BY FirstName) FROM Account WHERE Id IN :accountIds];
```

Comment:
Love the nested SOQL to pull Accounts and their related contacts in a single query